build.gradle

```diff
@@ -350,8 +366,12 @@ model {
     }
 }
 
-tasks.withType(CppCompile) { task ->
-    task.dependsOn project.tasks.nativeHeaders
+tasks.withType(CppCompile).configureEach {
+    dependsOn(project.tasks.nativeHeaders)
+    dependsOn(writeNativeVersionSources)
+    inputs.files(nativeHeadersDir)
+        .withPropertyName("nativeHeadersDir")
+        .withPathSensitivity(PathSensitivity.RELATIVE)
```

Comment:
Why do we manually have to add this as a dependency/input files using the runtime API? Shouldn't these be specified as input directories? And shouldn't we use the output of the tasks instead of declaring the dependencies separately?

Reply:
The directory doesn't know that the two files in the directory are generated by two separate tasks.

Reply:
So the inputs are now detected automatically by header analysis. It seems like the header analysis loses the task dependencies, since `CalculatedTaskInputFileCollection` doesn't override `visitDependencies`.